Tasks/AzureResourceGroupDeployment/models/DeployAzureRG.ts

```diff
@@ -106,14 +76,71 @@ export class AzureRGTaskParameters {
         }
     }
 
-    private getARMCredentials(connectedService: string): msRestAzure.ApplicationTokenCredentials {
+    private async getARMCredentials(connectedService: string): Promise<msRestAzure.ApplicationTokenCredentials> {
```

Comment:
Consider not using async, instead of await, you could try initializeAzureStackData.then